File: backend/src/main/java/com/barbershop/modules/image/service/ImageService.java
package com.barbershop.modules.image.service;

import java.io.IOException;
import org.springframework.core.io.Resource;
import org.springframework.web.multipart.MultipartFile;
import com.barbershop.modules.image.model.entity.Image;

public interface ImageService {
    Image uploadForOwner(Long ownerId, MultipartFile file) throws IOException;
    Image getOwnerImage(Long ownerId);
    void deleteOwnerImage(Long ownerId);
    Image updateOwnerImage(Long ownerId, MultipartFile file) throws IOException;
    Image uploadForBarber(Long barberId, MultipartFile file) throws IOException;
    Image uploadForCustomer(Long customerId, MultipartFile file) throws IOException;
    Image uploadForShop(Long shopId, MultipartFile file) throws IOException;
    Resource loadFileAsResource(String fileName) throws IOException;
    void deleteFileFromDisk(String fileName);
}

File: frontend/src/modules/image/services/image.service.ts
// @/modules/image/services/image.service.ts

import { ImageResponse } from "../types/image.types";

export interface ImageService {
  /**
   * Overwrite the single image for an owner
   */
  uploadOwnerImage(ownerId: number, file: File): Promise<ImageResponse>;

  /**
   * Get the single image for an owner
   */
  getOwnerImage(ownerId: number): Promise<ImageResponse | null>;

  uploadBarberImage(barberId: number, file: File): Promise<ImageResponse>;
  uploadCustomerImage(customerId: number, file: File): Promise<ImageResponse>;
  uploadShopImage(shopId: number, file: File): Promise<ImageResponse>;
  deleteOwnerImage(ownerId: number): Promise<void>;
}
